### PR TITLE
Merge NRP's August branch in preview

### PR DESCRIPTION
**Pull request by anton-evseev**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.csproj [hunk 1]
```diff
@@ -48,8 +48,12 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.Azure.Graph.RBAC.3.2.0-preview\lib\net45\Microsoft.Azure.Graph.RBAC.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Azure.Management.Authorization, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (praries880) [inline src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.csproj]: revert this change

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ApplicationGatewayTests.ps1 [hunk 2]
```diff
@@ -450,6 +451,7 @@ function Test-ApplicationGatewayCRUD2
 		$getgw = Set-AzureRmApplicationGatewaySslCertificate -ApplicationGateway $getgw -Name $sslCert01Name -CertificateFile $sslCert01Path -Password $pw01
 		Assert-NotNull $getgw.SslCertificates[0].Password
 
+		#[SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine")]
```
[the remark this answers is not in the document]
Comment (number213) [reply src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ApplicationGatewayTests.ps1]: Yes https://github.com/Azure/azure-powershell/blob/81b775330dfbaa8498c99b2b513247b5372a1aae/src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ApplicationGatewayTests.ps1#L481

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ApplicationGatewayTests.ps1 [hunk 3]
```diff
@@ -387,6 +387,7 @@ function Test-ApplicationGatewayCRUD2
 		$gipconfig = New-AzureRmApplicationGatewayIPConfiguration -Name $gipconfigname -Subnet $gwSubnet
 
 		# frontend part
+		#[SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine")]
```
[the remark this answers is not in the document]
Comment (number213) [reply src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ApplicationGatewayTests.ps1]: Yes, AppGw gets deleted Yes https://github.com/Azure/azure-powershell/blob/81b775330dfbaa8498c99b2b513247b5372a1aae/src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ApplicationGatewayTests.ps1#L481

File: src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/NewAzureServiceEndpointPolicyCommand.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Do not ask for confirmation if you want to overrite a resource")]
+        public SwitchParameter Force { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            WriteWarning("The output object type of this cmdlet will be modified in a future release.");
```
[the remark this answers is not in the document]
Comment (number213) [reply src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/NewAzureServiceEndpointPolicyCommand.cs]: talked with @chandrasekarsrinivasan, cmdlet's output is not going to be changed. Will remove this line

File: src/ResourceManager/Network/Commands.Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+            HelpMessage = "Do not ask for confirmation if you want to overrite a resource")]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            WriteWarning("The output object type of this cmdlet will be modified in a future release.");
```
[the remark this answers is not in the document]
Comment (number213) [reply src/ResourceManager/Network/Commands.Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs]: talked with @chandrasekarsrinivasan, cmdlet's output is not going to be changed. Will remove this line

**Comment by praries880** [comment]
on demand : https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/1073/

**Comment by praries880** [comment]
sign job : https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/1141/
